### ISSUE TITLE
RoPE MHA applies dropout at test time

### ISSUE THREAD

**Issue by pplantinga** [speechbrain/speechbrain#2888]
### Describe the bug

Dropout is always passed as a flat rate independent of training status.

https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/nnet/attention.py#L1367-L1374

### Expected behaviour

Dropout should be 0.0 when not in training mode.

### Relevant Log Output

```shell

```

**Comment by TParcollet** [comment]
@shucongzhang @pplantinga isn't that automatically managed by torch internally ?!?!?!?!

**Comment by pplantinga** [comment]
> @shucongzhang @pplantinga isn't that automatically managed by torch internally ?!?!?!?!

My understanding is that in general the nn.Module interface is stateful and manages things like dropout, but the functional interface is stateless and expects the caller to manage things like dropout

**Comment by Adel-Moumen** [comment]
> > [@shucongzhang](https://github.com/shucongzhang) [@pplantinga](https://github.com/pplantinga) isn't that automatically managed by torch internally ?!?!?!?!
> 
> My understanding is that in general the nn.Module interface is stateful and manages things like dropout, but the functional interface is stateless and expects the caller to manage things like dropout

I believe in what @pplantinga says. This is a small repro here: https://colab.research.google.com/drive/1q36wafwaifWWYjldMAisEdCj-OobUH1-?usp=sharing as you can see, running multiple times the same input leads to different outputs.

When adding:
`dropout_p=self.dropout if self.training else 0`
It solves the issue.

<img width="844" alt="Image" src="https://github.com/user-attachments/assets/39e75081-9b20-4702-9bc6-7a28ecc5a849" />
